jina/math/dimensionality_reduction.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+
+
+class PCA:
+    """A class for PCA dimensionality reduction in Jina."""
+
+    def __init__(self, n_components: int, whitening: bool = False):
+        self.n_components = n_components
+        self.whitening = whitening
+        self.e_values = None
+        self.w = None
```

Comment:
what is `w`? `coefficient vector`?

Reply:
These are the weights learned by PCA (eigenvectors as columns)